### PR TITLE
Server & GitHub Integration use Gradle Kotlin DSL

### LINKED ISSUE #969: Switch Gradle scripts to use Kotlin DSL
- #968 upgrades us to Gradle >= 5.0 where Kotlin DSL is officially stable and supported. 
- Kotlin DSL enables better autocompletion etc within Intellij.
- We're already using Kotlin everywhere else.
- As a bonus the ktlint plugin in #968 checks (and can format) the build scripts too.

(I got carried away with the #968 branch and have the additional change ready to go for the server, haven't touched the github integration yet)

### PR DESCRIPTION
**Pull request by roxspring**
- Builds on top of #968 
- Server & GitHub Integration Gradle scripts use Kotlin DSL
- Added & updated dependencies so that GitHub Integration builds under Java 11

Closes #969

### CODE REVIEW

File: server/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        maven("https://repo.spring.io/libs-release")
+    }
+}
+
+plugins {
+    kotlin("jvm").version("1.3.21")
+    kotlin("plugin.jpa").version("1.3.21")
+    kotlin("plugin.noarg").version("1.3.21")
+    kotlin("plugin.spring").version("1.3.21")
+    kotlin("plugin.allopen").version("1.3.21")
```
Comment (roxspring) [inline server/build.gradle.kts]: I'll extract a `var kotlinVersion = "1.3.21"`

File: server/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    src("http://json-schema.org/draft-04/schema")
+    dest("$rootDir/src/main/resources/schemas/json-schema.json")
+    onlyIfModified(true)
+}
+
+tasks.register("downloadSwaggerSchema", Download::class) {
+    src("http://swagger.io/v2/schema.json")
+    dest("$rootDir/src/main/resources/schemas/swagger-schema.json")
+    onlyIfModified(true)
+}
```
Comment (roxspring) [inline server/build.gradle.kts]: Making these changes I noticed that `src/main/resources/schemas/json-schema.json` is committed to the repo but `src/main/resources/schemas/swagger-schema.json` isn't, while both really come from an outside source and ought to be treated consistently.

- Should we keep a copy in the repo?
  - If so, is it still worth having the build script attempt to update from elsewhere??
- Or should we keep neither in the repo and only download - in which case we probably ought to `.gitignore` them too

File: server/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        kotlinOptions.jvmTarget = "1.8"
+    }
+}
+
+val versions = mapOf(
+        "jacoco" to "0.8.2"
+)
+
+dependencies {
+    val spring_boot_version = "2.0.4.RELEASE"
```
Comment (roxspring) [inline server/build.gradle.kts]: I'll rename the variable `springBootVersion` to be more Kotlin style.

File: server/build.gradle.kts (hunk 2 above)
Comment (maxim-tschumak) [reply server/build.gradle.kts]: I vote for keeping neither in the repo and downloading on the fly.

File: github-integration/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+tasks.jacocoTestReport {
+    reports {
+        xml.isEnabled = true
+    }
+}
+
+tasks.jar {
+    archiveBaseName.set("bark")
```
Comment (maxim-tschumak) [inline github-integration/build.gradle.kts]: Let's move away from "bark" to more functional i.e. `zally-github-integration`.

File: server/build.gradle.kts (hunk 2 above)
Comment (roxspring) [reply server/build.gradle.kts]: Done: 043cb41450049707294e2b0ad7615c56c9ec98e3

**Comment by roxspring** [comment]
- rebased on master
- addressed review comments

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by tkrop** [comment]
:+1:

**Pull request merged**